Content.Goobstation.Server/Xenobiology/XenobiologySystem.Actions.cs

```diff
@@ -0,0 +1,168 @@
+// SPDX-FileCopyrightText: 2025 August Eymann <august.eymann@gmail.com>
+// SPDX-FileCopyrightText: 2025 GoobBot <uristmchands@proton.me>
+// SPDX-FileCopyrightText: 2025 SolsticeOfTheWinter <solsticeofthewinter@gmail.com>
+// SPDX-FileCopyrightText: 2025 TheBorzoiMustConsume <197824988+TheBorzoiMustConsume@users.noreply.github.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using Content.Goobstation.Shared.Xenobiology;
+using Content.Goobstation.Shared.Xenobiology.Components;
+using Content.Shared._Shitmed.Targeting;
+using Content.Shared.Examine;
+using Content.Shared.Humanoid;
+using Content.Shared.Mobs;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Popups;
+using Robust.Shared.Containers;
+
+namespace Content.Goobstation.Server.Xenobiology;
+
+/// <summary>
+/// This handles any actions that slime mobs may have.
+/// </summary>
+public partial class XenobiologySystem
+{
+    private void InitializeActions()
+    {
+        SubscribeLocalEvent<SlimeLatchEvent>(OnLatch);
+
+        SubscribeLocalEvent<SlimeComponent, ComponentStartup>(OnComponentInit);
+        SubscribeLocalEvent<SlimeComponent, ExaminedEvent>(OnExamined);
+
+        SubscribeLocalEvent<SlimeComponent, EntRemovedFromContainerMessage>(OnEntityEscape);
+        SubscribeLocalEvent<SlimeComponent, MobStateChangedEvent>(OnEntityDied);
+
+        SubscribeLocalEvent<SlimeDamageOvertimeComponent, MobStateChangedEvent>(OnConsumedEntityDied);
+    }
+
+    private void UpdateHunger()
+    {
+        var query = EntityQueryEnumerator<SlimeDamageOvertimeComponent>();
+        while (query.MoveNext(out var uid, out var dotComp))
+        {
+            if (_gameTiming.CurTime < dotComp.NextTickTime
+                || _mobState.IsDead(uid))
+                continue;
+
+            var addedHunger = (float)dotComp.Damage.GetTotal();
+            dotComp.NextTickTime = _gameTiming.CurTime + dotComp.Interval;
+            _damageable.TryChangeDamage(uid, dotComp.Damage, ignoreResistances: true, ignoreBlockers: true,  targetPart: TargetBodyPart.All);
+
+            if (!TryComp<HungerComponent>(dotComp.SourceEntityUid, out var hunger) ||
+                dotComp.SourceEntityUid is not { } sourceEntity)
```

Comment:
Logic goes at the start of every line, makes readability easier on large lines.
```suggestion
            if (!TryComp<HungerComponent>(dotComp.SourceEntityUid, out var hunger)
                || dotComp.SourceEntityUid is not { } sourceEntity)